### DISCUSSION TITLE
使用LakeSoulTable.upsert()时不管创不创建表都提示表不存在

### DISCUSSION THREAD

**Discussion by yzb1234** [lakesoul-io/LakeSoul#131] [Q&A] [answered]
我创建了表也是提示表不存在

**Comment by dmetasoul01** [comment]
你好，调用的代码和报错的信息可以发一下么？

**Comment by yzb1234** [comment]
Exception in thread "main" org.apache.spark.sql.AnalysisException: Table s3://ccf-datalake-contest/datalake_table doesn't exist.


spark.sql(
      """
        |CREATE TABLE datalake_table (
        |   uuid string,
        |   ip string,
        |   hostname string,
        |   requests int,
        |   name string,
        |   city string,
        |   job string,
        |   phonenum string)
        | USING lakesoul
        |""".stripMargin)

    val lakeSoulTable = LakeSoulTable.forPath(tablePath)
    lakeSoulTable.upsert(dfSum)

**Comment by moresun** [comment, marked as answer]
sql(
          s"""
             |CREATE TABLE lakesoul_test(a LONG NOT NULL, b String)
             |USING lakesoul
             |OPTIONS('path'='${dir.getCanonicalPath}')""".stripMargin)
需要指定path，表会在指定path中存储数据，是不是漏掉了

        Seq((1, "a"), (2, "insert")).toDF("id", "change_kind")
          .write
          .mode("overwrite")
          .format("lakesoul")
          .option("lakesoul_cdc_change_column", "change_kind")
          .save(path)
也可以使用dataframe这种方式建表
代码案例可以参考如下case
https://github.com/meta-soul/LakeSoul/blob/main/lakesoul-spark/src/test/scala/org/apache/spark/sql/lakesoul/DDLSuite.scala

https://github.com/meta-soul/LakeSoul/blob/main/lakesoul-spark/src/test/scala/org/apache/spark/sql/lakesoul/TableCreationTests.scala

**Comment by dmetasoul01** [comment]
使用 SQL 建表时，默认的路径是 spark.sql.warehouse.dir 这个配置指定的，默认是当前运行本地目录下的 spark-warehouse 子目录。可以使用 path 选项或者 LOCATION 语法指定路径。
